client/src/charts/nivo/NivoCharts.js

```diff
@@ -0,0 +1,1037 @@
+        markers,
+      }}
+      layout = 'horizontal'
+      keys = {_.union([''],keys)}
+      labelFormat={_.isUndefined(label_format) ? null : label_format}
+      tooltip={ (d) => tooltip( [d], get_formatter(is_money, text_formatter, false) ) }
+      axisBottom={remove_bottom_axis ? null : bttm_axis}
+      axisTop={add_top_axis ? top_axis : null}
+      axisLeft={
+        remove_left_axis ?
+          null :
+          {
+            tickValues: tick_value || 6,
+            format: (d) => get_formatter(is_money, text_formatter)(d),
+            min: "auto",
+            max: "auto",
+            ...(left_axis || {}),
+          }
+      }
+      padding={padding}
+      borderColor="inherit:darker(1.6)"
+      motionDamping={motion_damping}
+      motionStiffness={motion_stiffness}
+      labelTextColor={window.infobase_color_constants.textColor}
+      labelSkipWidth={labelSkipWidth}
+    />;
+
+    return <InteractiveGraph graph={graph} table={table} />;
+  }
+};
+NivoResponsiveHBar.defaultProps = {
+  ...general_default_props,
+
+  bttm_axis: {
+    tickSize: 7,
+    tickPadding: 10,
+  },
+  theme: {
+    legends: {
+      text: {
+        fontSize: 14,
+      },
+    },
+    labels: {
+      text: {
+        fontSize: 14,
+      },
+    },
+  },
+  isInteractive: true,
+  max: "auto",
+  min: "auto",
+  padding: 0.3,
+  motion_damping: 15,
+  motion_stiffness: 95,
+  labelSkipWidth: 10,
+};
+
+export class NivoResponsiveLine extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      y_scale_zoomed: false,
+    };
+  }
+  render(){
+    const {
+      data,
+      is_money,
+      raw_data,
+      margin,
+      tick_amount,
+      colors,
+      colorBy,
+      max,
+      min,
+      enableArea,
+      enableGridX,
+      enableGridY,
+      left_axis,
+      disable_yaxis_zoom,
+      yScale,
+      motion_damping,
+      motion_stiffness,
+      enableDotLabel,
+      remove_bottom_axis,
+      remove_left_axis,
+      bttm_axis,
+      text_formatter,
+      stacked,
+      theme,
+      tooltip,
+      markers,
+      legends,
+      layers,
+      disable_table_view,
+    } = this.props;
+
+    const {
+      y_scale_zoomed,
+    } = this.state;
+
+    legends && (legends[0].symbolShape = fixedSymbolShape);
+
+    const table_data = _.map(data, row => ({
+      col_data: _.chain(row.data)
+        .map(d => [d.x,get_formatter(is_money, text_formatter, true, true)(d.y)])
+        .fromPairs()
+        .assign({label: row.id})
+        .value(),
+      label: row.id}));
+    const table_header_keys = _.concat(['label'], _.chain(data)
+      .map( d=>_.map( d.data, d=>d.x ) )
+      .flatten()
+      .uniq()
+      .value() );
+    
+    const table = !disable_table_view && <DisplayTable data={table_data} column_keys={table_header_keys} sort_keys={table_header_keys} table_data_headers={table_header_keys} table_name={"TODO"}/>;
+
+    const zoom_button = (!disable_yaxis_zoom && !enableArea) ?
```

Comment:
Could this still work written `(!disable_yaxis_zoom && !enableArea) &&` ?

Reply:
Sorry, I'm not sure what you mean here

Reply:
I just thought it would look cleaner short circuiting with && instead of ternary here like this:
```
    const zoom_button = (!disable_yaxis_zoom && !enableArea) &&
      <button
         ...
      </button>;
```
but not sure if it would break because `zoom_button` value would be `false` instead of `undefined`